src/Symfony/Component/TypeInfo/Tests/TypeResolver/ReflectionReturnTypeResolverTest.php

```diff
@@ -64,6 +64,10 @@ public function testResolve()
 
     public function testCreateTypeContextOrUseProvided()
     {
+        if (\PHP_VERSION_ID >= 80500) {
+            $this->markTestSkipped('self and parent types are resolved at compile time since PHP 8.5.');
```

Comment:
Why do we need to skip those tests ? Are they covering a case where we resolve them in a weird way ? If yes, maybe we should rewrite the test instead of cover a case where we resolve `self` properly (which would then also pass even if `self` is pre-resolved by PHP)